src/components/ImageSelector/ImageSelector.vue

```diff
@@ -422,31 +1733,61 @@ export default {
         "v-museum-20": this.speed === 50,
       };
     },
+
     selectArrayDisplay() {
       return {
         "v-start": this.isBeginning,
         "v-normal": !this.isBeginning && !this.isEnd,
         "v-end": this.isEnd,
       };
     },
-    ...mapState(["images", "isLoadingImage"]),
+    ...mapState([
+      "images",
+      "isLoadingImage",
+      "relatedImages",
+      "secondRelatedImages",
+      "completionData",
+    ]),
+    ...mapGetters({
+      getImagesByDecade: "getImagesByDecade",
+      getCompletionByDecade: "getCompletionByDecade",
+      getVisitedIndexByDecade: "getVisitedIndexByDecade",
+    }),
   },
   mounted() {
-    // Use to find the ratio and to add the content correctly in the scroll and to know the translation size
     const { width, height } = useWindowSize();
     this.windowHeight = height;
     this.windowWidth = width;
 
-    // The parameter for the year search will come from the previous selection view.
-    // Currently, this value is hard-coded for testing purpose.
-    this.$store.dispatch("initializeCarousel", {
-      decade: "193",
-    });
+    this.rectangleHeight = this.thumbnailHeight() + 20;
+    this.rectangleWidth = this.thumbnailWidth() + 20;
+
+    this.currentXPosition = this.defineLeftPositionCenterPage();
+    this.currentYPosition = this.defineTopPositionCenterPage();
+
+    this.currentDecade = this.$route.params.decade;
+    const data = this.getImagesByDecade(this.currentDecade);
+    const id = this.$route.query.id;
+    const comeFromOnboarding = this.$route.query.comeFromOnboarding;
+
+    if (id) {
+      this.loadFromFocusPage(data, id);
+    } else if (comeFromOnboarding) {
+      this.loadFromOnboarding(data);
+    } else if (data === undefined) {
+      this.$store.dispatch("initializeCarousel", {
+        decade: this.currentDecade,
+      });
+    } else {
+      this.data = data.data;
+      this.loadInitialData();
+    }
   },
 };
 </script>
 
 <style scoped>
 @import "./imageselector.css";
 @import "./sliderspeed.css";
+@import "./carouselAnimation.css";
 </style>
```

Comment:
this file is probably too big and should ideally either be splitted in different components (completion spiral could be I guess a specific vue component to reuse everywhere).

Also would need some refactoring, hard to follow and probably hard to debug for the different position computations with multiple parameters.
I would not do it now since too risky but to keep in mind if we work again on that

Reply:
Yes, I think too that a refactoring is necessary